app/models/smart_proxy.rb

```diff
@@ -64,6 +64,10 @@ def self.smart_proxy_ids_for(hosts)
     ids.flatten.compact.map { |i| i.to_i }.uniq
   end
 
+  def hosts_count
+    Host::Managed.search_for("smart_proxy = #{name}").count
```

Comment:
It could technically be done in one query(i.e. counting the first query in the scoped search), but the second query executed is only a count() so it should be very fast (the filtering can be done in the index as we only filter on id and type).